solvers/reweighted.py

```diff
@@ -0,0 +1,59 @@
+from benchopt import BaseSolver, safe_import_context
+
+with safe_import_context() as import_ctx:
+    import numpy as np
+    from numba import njit
+    from flashcd.estimators import WeightedLasso
+
+
+if import_ctx.failed_import:
+
+    def njit(f):  # noqa: F811
+        return f
+
+
+@njit
+def gprime(x, lmbd, gamma):
+    return np.maximum(
+        0, lmbd - np.abs(x) / gamma)
+
+
+class Solver(BaseSolver):
+    name = "reweighted"
+    install_cmd = "conda"
+    requirements = ["numba"]
+    references = [
+        'E. J. Candès, M. B. Wakin, S. P. Boyd, '
+        "Enhancing Sparsity by Reweighted l1 Minimization"
+        "Journal of Fourier Analysis and Applications,"
+        "vol. 14, pp. 877-905 (2008)"
+    ]
+
+    def set_objective(self, X, y, lmbd, gamma):
+        # use Fortran order to compute gradient on contiguous columns
+        self.X, self.y = np.asfortranarray(X), y
+        self.lmbd, self.gamma = lmbd, gamma
+
+        # Make sure we cache the numba compilation.
+        self.run(1)
+
+    def run(self, n_iter):
```

Comment:
@mathurinm On donne quoi au n_iter pour benchopt du coup ? Les iters de la resolution des weighted?

Reply:
yes as a first try this makes the most sense